src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1465,7 +1465,7 @@ public override ISymbol GetDeclaredSymbol(BasePropertyDeclarationSyntax declarat
         /// <returns>The symbol that was declared.</returns>
         public override IPropertySymbol GetDeclaredSymbol(PropertyDeclarationSyntax declarationSyntax, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return (IPropertySymbol)GetDeclaredMemberSymbol(declarationSyntax);
+            return ((PropertySymbol)GetDeclaredMemberSymbol(declarationSyntax)).GetPublicSymbol();
```

Comment:
A convenience thing we could consider adding to `GetPublicSymbol()`: a `T GetPublicSymbol<T>() where T : ISymbol` variant, to avoid having to insert all of these parens to contain casting. Just a suggestion, feel free to ignore if you don't want to take it.

Reply:
> A convenience thing we could consider adding to GetPublicSymbol(): a T GetPublicSymbol<T>() where T : ISymbol variant, to avoid having to insert all of these parens to contain casting. Just a suggestion, feel free to ignore if you don't want to take it.

I explicitly hide this API so that it wasn't easy to "cast" to an arbitrary ISymbol type using a non-obvious way. A cast should be a cast syntactically, then there is no confusion to the reader. 



---
In reply to: [339204675](https://github.com/dotnet/roslyn/pull/39498#discussion_r339204675) [](ancestors = 339204675)